pkg/aws_client/vpc_endpoint.go

```diff
@@ -72,6 +73,39 @@ func (c *AWSClient) SelectVPCForVPCEndpoint(ctx context.Context, ids ...string)
 	return minVpcId, nil
 }
 
+// FilterVpcIdsByTags tags in a list of tags and returns a list of AWS VPC Ids that have all of the provided tags
+func (c *AWSClient) FilterVpcIdsByTags(ctx context.Context, tags []avov1alpha2.Tag) ([]string, error) {
+	if len(tags) == 0 {
+		return nil, errors.New("must specify tags when filtering VPCs by tags")
+	}
+
+	filters := make([]types.Filter, len(tags))
+	for i, tag := range tags {
+		filters[i] = types.Filter{
+			Name:   aws.String(fmt.Sprintf("tag:%s", tag.Key)),
+			Values: []string{tag.Value},
+		}
+	}
+
+	resp, err := c.ec2Client.DescribeVpcs(ctx, &ec2.DescribeVpcsInput{
+		Filters: filters,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(resp.Vpcs) == 0 {
```

Comment:
curious, shouid we ever hit this if `case len(vpce.Spec.Vpc.Tags) > 0:` is called before this function?